web/src/features/rbac/server/membersRouter.ts

```diff
@@ -0,0 +1,818 @@
+                  not: Role.NONE,
+                },
+              },
+              {
+                projectId: query.projectId,
+                projectRole: {
+                  not: Role.NONE,
+                },
+              },
+            ],
+          }
+        : {}),
+    },
+    include: {
+      invitedByUser: {
+        select: {
+          name: true,
+          image: true,
+        },
+      },
+    },
+    orderBy: {
+      createdAt: "desc",
+    },
+    take: query.limit,
+    skip: query.page * query.limit,
+  });
+
+  const totalCount = await prisma.membershipInvitation.count({
+    where: {
+      orgId: query.orgId,
+    },
+  });
+
+  return {
+    invitations: invitations.map((i) => ({
+      ...i,
+    })),
+    totalCount,
+  };
+}
+
+export const membersRouter = createTRPCRouter({
+  allFromOrg: protectedOrganizationProcedure
+    .input(orgLevelMemberQuery)
+    .query(async ({ input, ctx }) => {
+      throwIfNoOrganizationAccess({
+        session: ctx.session,
+        organizationId: input.orgId,
+        scope: "organizationMembers:read",
+      });
+      return getMembers(ctx.prisma, input);
+    }),
+  allFromProject: protectedProjectProcedure
+    .input(projectLevelMemberQuery)
+    .query(async ({ input, ctx }) => {
+      const orgAccess = hasOrganizationAccess({
+        session: ctx.session,
+        organizationId: input.orgId,
+        scope: "organizationMembers:read",
+      });
+      const projectAccess = hasProjectAccess({
+        session: ctx.session,
+        projectId: input.projectId,
+        scope: "projectMembers:read",
+      });
+      if (!orgAccess && !projectAccess) {
+        throw new TRPCError({
+          code: "FORBIDDEN",
+          message: "You do not have the required access rights",
+        });
+      }
+      return getMembers(ctx.prisma, input, orgAccess);
+    }),
+  allInvitesFromOrg: protectedOrganizationProcedure
+    .input(orgLevelInviteQuery)
+    .query(async ({ input, ctx }) => {
+      throwIfNoOrganizationAccess({
+        session: ctx.session,
+        organizationId: input.orgId,
+        scope: "organizationMembers:read",
+      });
+      return getInvites(ctx.prisma, input);
+    }),
+  allInvitesFromProject: protectedProjectProcedure
+    .input(projectLevelInviteQuery)
+    .query(async ({ input, ctx }) => {
+      const orgAccess = hasOrganizationAccess({
+        session: ctx.session,
+        organizationId: input.orgId,
+        scope: "organizationMembers:read",
+      });
+      const projectAccess = hasProjectAccess({
+        session: ctx.session,
+        projectId: input.projectId,
+        scope: "projectMembers:read",
+      });
+      if (!orgAccess && !projectAccess) {
+        throw new TRPCError({
+          code: "FORBIDDEN",
+          message: "You do not have the required access rights",
+        });
+      }
+      return getInvites(ctx.prisma, input, orgAccess);
+    }),
+  create: protectedOrganizationProcedure
```

Comment:
this function is super complex. I would have split this up into two -> one for project, one for org.

Reply:
disagree, one-by-one has access checks